### PR TITLE
Adding support for SSL dual mode (#712)

### PR DESCRIPTION
**Pull request by sachetalva**
* Adding SSL dual mode support for SSLOnly

(cherry picked from commit 27f33fb)

Co-authored-by: Vengadanathan Srinivasan <vengadan@amazon.com>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/security/test/SingleClusterTest.java [hunk 1]
```diff
@@ -42,8 +42,8 @@
 
 public abstract class SingleClusterTest extends AbstractSecurityUnitTest {
 
-    private static final int DEFAULT_MASTER_NODE_NUM = 3;
```
Comment (hardik-k-shah) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/test/SingleClusterTest.java]: why this change is specific to 0.9 branch? 

I see in other versions it is 
private static final int DEFAULT_MASTER_NODE_NUM = 3;
private static final int DEFAULT_FIRST_DATA_NODE_NUM = 2;

Comment (sachetalva) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/test/SingleClusterTest.java]: This numbering is with respect to DEFAULT_MASTER_WITHOUT_SECURITY_PLUGIN and DEFAULT_ONE_DATA_NODE_WITHOUT_SECURITY_PLUGIN defined in ClusterConfiguration.
These enums define a 3 node cluster with 1 node not having the security plugin.
in ES 6.x cluster the node numbering seems to be in a reverse order compared to ES 7.x cluster, which why had to do this change for ES 6.x plugins

**Review by hardik-k-shah — APPROVED** [review]
